Sources/Private/Experimental/Animations/CALayer+addAnimation.swift

```diff
@@ -39,11 +39,48 @@ extension CALayer {
     for property: LayerProperty<ValueRepresentation>,
     keyframes: ContiguousArray<Keyframe<KeyframeValue>>,
     value keyframeValueMapping: (KeyframeValue) -> ValueRepresentation,
-    context: LayerAnimationContext)
+    context: LayerAnimationContext,
+    applyDirectlyToLayerIfPossible: Bool = true)
     -> CAPropertyAnimation?
   {
     guard !keyframes.isEmpty else { return nil }
 
+    // If there is exactly one keyframe value, we can improve performance
+    // by applying that value directly to the layer instead of creating
+    // a relatively expensive `CAKeyframeAnimation`.
+    if
+      keyframes.count == 1,
+      applyDirectlyToLayerIfPossible
+    {
+      let keyframeValue = keyframeValueMapping(keyframes[0].value)
+
+      // If the keyframe value is the same as the layer's default value for this property,
+      // then we can just ignore this set of keyframes.
+      if keyframeValue == property.defaultValue {
+        return nil
+      }
+
+      // If the property on the CALayer being animated hasn't been modified from the default yet,
+      // then we can apply the keyframe value directly to the layer using KVC instead
+      // of creating a `CAAnimation`. (For example, this check prevents us from using
+      // KVC to change the `anchorPoint`, which conflicts with where we set the `anchorPoint`
+      // for our layers to a non-default value in `BaseAnimationLayer` / `fillBoundsOfSuperlayer()`)
+      let currentValue = value(forKey: property.caLayerKeypath) as? ValueRepresentation
```

Comment:
dropping a note to remember to check this before merging -- since KVC lookups are somewhat expensive, we may want to check if `property.defaultValue` was defined (e.g. is not `nil`) before doing the KVC lookup. I think that shouldn't affect any runtime behaviors, but would just lets us avoid KVC lookups for conditions that are going to fail anyway.

Reply:
cool, I made this change in https://github.com/airbnb/lottie-ios/pull/1518/commits/e1f7a758d7fee38fd8b8e014a114b30688559ab4 and it improved performance a bit more